AppCenterAnalytics/AppCenterAnalytics/Internals/Model/Properties/MSBooleanTypedProperty.m

```diff
@@ -0,0 +1,49 @@
+#import "MSBooleanTypedProperty.h"
+#import "MSConstants+Internal.h"
+
+@implementation MSBooleanTypedProperty
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    self.type = @"boolean";
+  }
+  return self;
+}
+
+- (instancetype)initWithCoder:(NSCoder *)coder {
+  self = [super initWithCoder:coder];
+  if (self) {
+    _value = [coder decodeBoolForKey:kMSTypedPropertyValue];
+  }
+  return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder {
+  [super encodeWithCoder:coder];
+  [coder encodeBool:self.value forKey:kMSTypedPropertyValue];
+}
+
+/**
+ * Serialize this object to a dictionary.
+ *
+ * @return A dictionary representing this object.
+ */
+- (NSMutableDictionary *)serializeToDictionary {
+  NSMutableDictionary *dict = [super serializeToDictionary];
+  dict[kMSTypedPropertyValue] = @(self.value);
```

Comment:
Does it work? We had an issue when we serialize BOOL to dictionary, we have used below pattern.
`dict[kMSTypedPropertyValue] = self.value ? @YES : @NO`
Maybe **@(** ... **)** does the thing?

Reply:
Yep that makes it an `NSNumber` (so same as `@YES`)